File: backend/app/app/schemas/node.py
from datetime import datetime
from pydantic import BaseModel
from typing import Optional, List

# Shared properties
class NodeBase(BaseModel):
    parent_id: Optional[int]
    node_type: Optional[str]
    name: Optional[str]
    is_active: Optional[bool] = True


# Properties to receive on node creation
class NodeCreate(NodeBase):
    node_type: str
    name: str


# Properties to receive on node update
class NodeUpdate(NodeBase):
    pass


# Properties shared by models stored in DB
class NodeInDBBase(NodeBase):
    id: int
    depth: int
    created_at: datetime
    updated_at: datetime
    created_by_id: int
    updated_by_id: int

    class Config:
        orm_mode = True


# Properties to return to client
class Node(NodeInDBBase):
    pass


class NodeList(BaseModel):
    total_records: int
    nodes: Optional[List[Node]] = []


# Properties properties stored in DB
class NodeInDB(NodeInDBBase):
    pass


class NodeChild(BaseModel):
    node_id: int
    child_type: str
    child_id: int
    child_name: str

File: frontend2/src/interfaces/node.ts
export interface INode {
  id: number;
  parent_id?: number;
  node_type: string;
  name: string;
  is_active: boolean;
  depth: number;
  created_at: Date;
  updated_at: Date;
  created_by_id: number;
  updated_by_id: number;
}

export interface INodeList {
  total_records: number;
  nodes: INode[];
}

export interface INodeCreate {
  node_type: string;
  name: string;
  parent_id?: number;
}

export interface INodeUpdate {
  parent_id?: number;
  node_type?: string;
  name?: string;
  is_active?: boolean;
}

export interface INodeChild {
  node_id: number;
  child_type: string;
  child_id: number;
  child_name: string;
}
